include/vrt.h

```diff
@@ -456,14 +461,6 @@ VCL_BACKEND VRT_DirectorResolve(VRT_CTX, VCL_BACKEND);
 void VRT_acl_log(VRT_CTX, const char *);
 int VRT_acl_match(VRT_CTX, VCL_ACL, VCL_IP);
 
-/***********************************************************************
```

Comment:
:-1:  on removing `VRT_re_*` from VRT.
This will break vmods building VREs dynamically.
According to my vmod library, this issue is real.

Reply:
In that case you are asking for a panic if the dynamically-built regex has a syntax error.

FWIW you have `vre.h` to work with such regular expressions. That's what vmod_cookie was using and it was not affected by the privatization of that part of the API. I would rather reconsider how those VMODs are operating today.

Unless we don't want to offer VCL scope and compile-time checks to the REGEX type.

Reply:
the panic can be avoiding by another `VRE_compile()` before the `VRT_re_init()`, but I had a closer look and now agree that the use cases in vmods are [either wrong](https://github.com/varnish/varnish-modules/blob/f26e057727c6e4ff3c042dae19cfd64e75a95236/src/vmod_header.c#L254) or [motivated by convenience](https://github.com/Dridi/libvmod-querystring/blob/06f8206c91842d44f881c1a343c264312d3a36a7/src/vmod_querystring.c#L598) (examples chosen randomly ;) )

So I agree with you now.